crates/sparrow-scheduler/src/queue.rs

```diff
@@ -49,25 +51,41 @@ impl<T> GlobalQueue<T> {
     }
 
     pub(crate) fn push(&self, item: T) {
+        debug_println!(
+            crate::DEBUG_SCHEDULER,
+            "Pushing {item:?} for global execution"
+        );
         self.queue.push(item)
     }
 }
 
-impl<T> LocalQueue<T> {
+impl<T: std::fmt::Debug> LocalQueue<T> {
     /// Pop an item from the local queue, or steal from the global and sibling queues if it is empty.
     pub fn pop(&mut self) -> Option<T> {
         self.queue.pop()
     }
 
     pub(crate) fn push(&mut self, item: T) {
+        debug_println!(
+            crate::DEBUG_SCHEDULER,
+            "Pushing {item:?} for local execution"
+        );
         self.queue.push(item)
     }
 
     pub(crate) fn push_yield(&mut self, item: T) {
+        debug_println!(
+            crate::DEBUG_SCHEDULER,
+            "Pushing {item:?} for (deferred) local execution"
+        );
         self.queue.push_yield(item)
     }
 
     pub(crate) fn push_global(&self, item: T) {
+        debug_println!(
```

Comment:
I can probably delete these... it was for debugging some very specific things (why we were immediately ending, which turned out to just be because the worker loop immediately ended since the queue was empty)